ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -453,88 +499,86 @@ List<CloudBlobMetadata> getDeadBlobs(String partitionPath, String fieldName, lon
     queryRequestOptions.setPartitionKey(new PartitionKey(partitionPath));
     queryRequestOptions.setResponseContinuationTokenLimitInKb(continuationTokenLimitKb);
 
-    try {
-      List<CloudBlobMetadata> deadBlobsList = new ArrayList<>();
-      AtomicReference<Double> requestCharge = new AtomicReference<>(0.0);
-
-      // Execute cosmos query
-      CosmosPagedFlux<CloudBlobMetadata> pagedFluxResponse =
-          executeCosmosQuery(sqlQuerySpec, queryRequestOptions, azureMetrics.deadBlobsQueryTime);
-
-      //Set the maximum entries to be as returned in each page. There seems to be no way to set maximum number of entries across all pages.
-      pagedFluxResponse.byPage(maxEntries).flatMapSequential(fluxResponse -> {
```

Comment:
i am not familiar the difference between flatMapSequantial and subscribe methods, I will leave this part to ankur for review.